[THORN-2126] upgrade to WildFly 14

Ladicek opened the pull request:
- [x] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [x] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [ ] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [x] Have you built the project locally prior to submission with `mvn clean install`?

sberyozkin committed:
[THORN-2207] Initial version and artifactId updates

sberyozkin committed:
Changing the CDI literal package

sberyozkin committed:
[THORN-2207] Getting the core/container compiling

sberyozkin committed:
[THORN-2207] Getting the container parser test passing

sberyozkin committed:
Re-enabling the gradle plugins

sberyozkin committed:
[THORN-2207] Updating the nosql feature pack version as recommended at THORN-2126

sberyozkin committed:
Getting MP Health and monitor fractions compiling

sberyozkin committed:
Fixing the checkstyle issue

sberyozkin committed:
[THORN-2210] Initializing DeferredExtensionContext

sberyozkin committed:
[THORN-2218] Fixing Infinispan related compilation issues for now

sberyozkin committed:
[THORN-2210] Adding a WF14 comment

sberyozkin committed:
Fixing the mod_cluster compilation failure

sberyozkin committed:
Restoring the original imprt order

sberyozkin committed:
Removing the trailing comments

Ladicek committed:
[THORN-2210] fix creating DeferredExtensionContext

Ladicek committed:
add TODO comments to 3 places where cache eviction used to be configured, but was removed to make fractions compile

sberyozkin committed:
Fixing the import package names in the messaging fraction

sberyozkin committed:
Adding a WF14 comment to Infinispan fraction

Ladicek committed:
[THORN-2219] update JBoss Modules and also our overrides

Ladicek committed:
[THORN-2210] couple of changes to finalize the core upgrade

Ladicek committed:
a bunch of uncategorized work-in-progress changes to make the Maven build not fail

sberyozkin committed:
[THORN-2205] Create Jsonb and Jaxrs Jsonb fractions

sberyozkin committed:
Using module.conf instead of module.xml

sberyozkin committed:
Restoring and replacing Infinispan local cache eviction config with the objectMemory one

sberyozkin committed:
Setting the original KC cache eviction props correctly

Ladicek committed:
follow-up to the Camel stuff, add one module definition that is now missing with WildFly 14

Ladicek committed:
add JSON-B module to pom.xml

sberyozkin committed:
[THORN-2239] Add a new ee-security fraction and tests

sberyozkin committed:
[THORN-2227] Switching to microprofile-config-wildfly

Ladicek committed:
[THORN-2249] fix the JSP tests

Ladicek committed:
[THORN-2271] add workaround for WFLY-11373 to our integration of SmallRye Fault Tolerance

Ladicek committed:
[THORN-2272] fix how Infinispan caches are forced to be active

Ladicek committed:
[THORN-2249] fix the Management Console tests

Ladicek committed:
[THORN-2249] fix the mod_cluster tests

Ladicek committed:
[THORN-2249] add Weld module dependencies to MP Metrics to avoid Weld classloading warnings

Ladicek committed:
use existing String constants instead of literals in the EE fraction

Ladicek committed:
[THORN-2273] don't let RESTEasy provide their own MicroProfile RestClient

Ladicek committed:
[THORN-2249] avoid Maven warnings and duplicate uberjar packaging in HowTo builds

Ladicek committed:
[THORN-2274] upgrade Keycloak to 4.6.0.Final (#15)

sberyozkin committed:
[THORN-2221] Update all nosql config apis to 1.5.0 (#16)

sberyozkin committed:
[THORN-2265] Update microprofile-config-wildfly version to 1.5.1 (#17)

sberyozkin committed:
Update the spec number in the JSF fraction README (#18)

Ladicek committed:
[THORN-2249][THORN-2265] avoid using deprecated APIs and cleanup build to avoid Maven warnings

Ladicek committed:
[THORN-2281] fix how web.xml and jboss-deployment-structure.xml are assembled

Ladicek committed:
[THORN-2249] fix Arquillian module dependencies to avoid CDI warnings in some tests

sberyozkin committed:
[THORN-2226] Removig WF14 comments in the Java source (#21)

* [THORN-2226] Removig WF14 comments in the Java source

* [THORN-2226] Using the correct temp file prefix

sberyozkin committed:
[THORN-2226] Removing the unused property in MavenArtifactUtil (#22)

Ladicek committed:
[THORN-2250] get rid of MP Health warnings in standalone test suite

Ladicek committed:
[THORN-2237] bump the fraction plugin to a released version

Ladicek committed:
[THORN-2272] fix how "hibernate" caches are forcefully activated

The `hibernate` cache container doesn't define a default cache, so creating
an artifical service dependency on `default-cache.hibernate` can't work.
So for `hibernate`, we'll use the old approach of adding a service dependency
on `cache-container-configuration.hibernate`; for other cache containers,
we'll keep using the new approach.

Ladicek committed:
[THORN-2249] avoid Weld warnings in the CDI test suite

Ladicek committed:
[THORN-2282] upgrade Keycloak to 4.7.0.Final

Ladicek committed:
[THORN-2230] replace "swarm.arquillian.daemon.port" in all arquillian.xml-s by "thorntail.arquillian.daemon.port"

Ladicek committed:
[THORN-2271] better workaround for WFLY-11373 that doesn't cause MP FT TCK to fail

Ladicek committed:
[THORN-2215] update default configuration of the Undertow, Elytron and Infinispan fractions to match WildFly 14 subsystems

Ladicek committed:
[THORN-2207] finish updating dependencies to align with WildFly 14.0.1

Ladicek committed:
[THORN-2271] even better workaround for WFLY-11373

manovotn commented:
@Ladicek great job with WFLY-11373 workaround :)

Ladicek commented:
Thanks @manovotn, though I kinda hoped I'd find a `Literal` for `@Resource` somewhere ;-)

Ladicek commented:
Also could someone cleanup the disk on the Windows Jenkins agent? :-)

sberyozkin reviewed (commented):
All of these changes look good to me. @kenfinnigan, please provide your review

kenfinnigan commented on docs/howto/pom.xml:
```diff
@@ -88,18 +91,6 @@
       </resource>
     </resources>
     <plugins>
-      <plugin>
```
Why was this removed?

Ladicek replied:
Because each howto defines `thorntail-maven-plugin` on its own -- unfortunately, using a different `<id>` of the `<execution>`. So the Thorntail Maven plugin's `package` goal used to be invoked twice.

kenfinnigan replied:
Makes sense, thanks

kenfinnigan commented on fractions/javaee/jpa/pom.xml:
```diff
@@ -51,6 +51,11 @@
       <groupId>io.thorntail</groupId>
       <artifactId>ee</artifactId>
     </dependency>
+    <!-- WF14 -->
```
There's a hard dependency on Infinispan fraction for JPA? That's not good

sberyozkin replied:
This is because of https://github.com/wildfly/wildfly/blob/14.0.1.Final/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/as/jpa/main/module.xml#L49 I believe

Ladicek replied:
The Hibernate+Infinispan integration has moved from Hibernate to Infinispan, that might be the reason. But I'll look deeper.

Ladicek replied:
Unfortunately it seems it really is required. Otherwise I'm getting

```
    "WFLYCTL0412: Required services that are not installed:" => ["org.wildfly.clustering.infinispan.cache-configuration.hibernate.entity"],
    "WFLYCTL0180: Services with missing/unavailable dependencies" => [
        "jboss.persistenceunit.\"CdiJpaJtaTest.jar#MyPU\".__FIRST_PHASE__ is missing [org.wildfly.clustering.infinispan.cache-configuration.hibernate.entity]",
        "jboss.persistenceunit.\"CdiJpaJtaTest.jar#MyPU\" is missing [org.wildfly.clustering.infinispan.cache-configuration.hibernate.entity]"
    ]
```

That is, the persistence unit service (from the JPA subsystem) depends on the Hibernate cache services (from the Infinispan subsystem).

Not sure how it worked before.

kenfinnigan replied:
Well that's going to blow up uber jar sizes.

Oh well

Ladicek replied:
It does :-( The `jpa-jaxrs-cdi-jta` example has 110 MB with 2.2.1 and 126 MB now.

kenfinnigan commented on build-parent/pom.xml:
```diff
@@ -248,6 +252,42 @@
         <version>${version.org.jboss.shrinkwrap.descriptors}</version>
       </dependency>
 
+      <dependency>
```
Why are all these aether dependencies needed here? I didn't see them added anywhere

Ladicek replied:
Hmm, I don't recall adding this. I'll check.

Ladicek replied:
@sberyozkin It seems you added this, do you recall why?

Ladicek replied:
Got it. Previously, Eclipse Aether was managed in `wildfly-core-parent` (which we import in `<dependencyManagement>`), but that's no longer the case. So we have to manage it ourselves.

sberyozkin replied:
Yes, that is exactly why I added them, sorry, forgot I did it

kenfinnigan replied:
Ok, thanks for clarification

Ladicek committed:
[THORN-2226] review and address the remaining unfinished WF14 pieces

Ladicek commented:
Pushed a commit that addresses the remaining WF14 pieces. Unfortunately, it seems JPA having a hard dependency on Infinispan is now needed.

Ladicek committed:
[THORN-2226] split all Elytron permissions into individual fractions

Ladicek committed:
[THORN-2273] improve the solution to not let RESTEasy provide their own MP RestClient

Apparently, something fraction-like outside of the `fractions` directory
also has a module dependency on `org.jboss.resteasy.resteasy-jaxrs`.
This commit moves the `module-rewrite.conf` to the top-level directory,
so all fractions and fraction-likes obey the same rules.

Ladicek committed:
fix HowTo tests by managing Guava to an older version

Ladicek commented:
Linux build is green, yay! It would be great if we could get the disk cleaned up on the Windows agent, as getting a Windows build would be good for a change of this size.

kenfinnigan commented:
retest this please
